packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,865 @@
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function incrementActiveVotes(address group, address account, uint256 value) private {
+    ActiveVotes storage active = votes.active;
+    active.total = active.total.add(value);
+
+    uint256 unitsDelta = getActiveVotesUnitsDelta(group, value);
+
+    GroupActiveVotes storage groupActive = active.forGroup[group];
+    groupActive.total = groupActive.total.add(value);
+
+    groupActive.totalUnits = groupActive.totalUnits.add(unitsDelta);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].add(unitsDelta);
+  }
+
+  /**
+   * @notice Decrements the number of active votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function decrementActiveVotes(address group, address account, uint256 value) private {
+    ActiveVotes storage active = votes.active;
+    active.total = active.total.sub(value);
+
+    uint256 unitsDelta = getActiveVotesUnitsDelta(group, value);
+
+    GroupActiveVotes storage groupActive = active.forGroup[group];
+    groupActive.total = groupActive.total.sub(value);
+
+    groupActive.totalUnits = groupActive.totalUnits.sub(unitsDelta);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(unitsDelta);
+  }
+
+  /**
+   * @notice Returns the delta in active vote denominator for `group`.
+   * @param group The address of the validator group.
+   * @param value The number of active votes being added.
+   * @return The delta in active vote denominator for `group`.
+   */
+  function getActiveVotesUnitsDelta(address group, uint256 value) private view returns (uint256) {
+    // Preserve unitsDelta * total = value * totalUnits
+    return value.mul(votes.active.forGroup[group].totalUnits.add(1)).div(
+      votes.active.forGroup[group].total.add(1)
+    );
+  }
+
+  /**
+   * @notice Returns the groups that `account` has voted for.
+   * @param account The address of the account casting votes.
+   * @return The groups that `account` has voted for.
+   */
+  function getGroupsVotedForByAccount(address account) external view returns (address[] memory) {
+    return votes.groupsVotedFor[account];
+  }
+
+  /**
+   * @notice Deletes an element from a list of addresses.
+   * @param list The list of addresses.
+   * @param element The address to delete.
+   * @param index The index of `element` in the list.
+   */
+  function deleteElement(address[] storage list, address element, uint256 index) private {
+    // TODO(asa): Move this to a library to be shared.
+    require(index < list.length && list[index] == element);
+    uint256 lastIndex = list.length.sub(1);
+    list[index] = list[lastIndex];
+    delete list[lastIndex];
```

Comment:
after checking solidity docs, this step is unneeded

Reply:
Sweet! Didn't know this